src/pkg/services/m365/api/access.go

```diff
@@ -51,7 +49,7 @@ func (c Access) GetToken(
 
 	resp, err := c.Post(ctx, rawURL, headers, body)
 	if err != nil {
-		return graph.Stack(ctx, err)
+		return clues.Stack(err)
```

Comment:
this may also need to add clues if the middleware doesn't handle it

Reply:
Clients could be expected to add clues themselves in these cases.

Reply:
what's the rule for adding clues then? So far it seems to be "if the caller takes ctx and is a corso function assume it adds clues." This makes it seem like every dev is expected to read the called function to figure out if they need to add clues or not?

Reply:
> This makes it seem like every dev is expected to read the called function to figure out if they need to add clues or not?

I prefer to defer that to the writer, not the reader.  As in, every dev is expected to add clues if the func/internal package they're adding accepts a ctx, and the func they call does not.  If we uphold that, implementers don't need to go looking at the next func.  They can evaluate purely based on whether they're calling an internal or external package/client.